test/unit-test/mqtt_agent_utest.c

```diff
@@ -853,3 +904,215 @@ void test_MQTTAgent_Terminate_success( void )
     TEST_ASSERT_NULL( command.pArgs );
     TEST_ASSERT_EQUAL_PTR( stubCompletionCallback, command.pCommandCompleteCallback );
 }
+
+/**
+ * @brief Test MQTTAgent_CommandLoop behaviour when sent command is NULL.
+ */
+void test_MQTTAgent_CommandLoop_sent_command_null( void )
+{
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+    MQTTAgentCommandFuncReturns_t pReturnFlags;
+
+    setupAgentContext( &mqttAgentContext );
+
+    mqttAgentContext.mqttContext.connectStatus = MQTTConnected;
+    pReturnFlags.addAcknowledgment = false;
+    pReturnFlags.runProcessLoop = true;
+    pReturnFlags.endLoop = true;
+
+    MQTTAgentCommand_ProcessLoop_ExpectAnyArgsAndReturn( MQTTBadParameter );
+    MQTTAgentCommand_ProcessLoop_ReturnThruPtr_pReturnFlags( &pReturnFlags );
+
+    mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
+
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+}
+
+/**
+ * @brief Test MQTTAgent_CommandLoop behaviour when sent command is not NULL.
+ */
+void test_MQTTAgent_CommandLoop_with_sent_command( void )
```

Comment:
This name doesn't really tell me what it's testing. It looks like this test is to ensure that a completion callback is invoked when a command is processed?

Reply:
This test is to cover the following branch if part
if( pCommand != NULL )
{
assert( pCommand->commandType < NUM_COMMANDS );
commandFunction = pCommandFunctionTable[ pCommand->commandType ];
pCommandArgs = pCommand->pArgs;
}
else
{
commandFunction = pCommandFunctionTable[ NONE ];
}